language/move-prover/interpreter/src/concrete/player.rs

```diff
@@ -2112,7 +2113,7 @@ impl<'env> FunctionContext<'env> {
             .into_iter()
             .map(|e| e.into_u8())
             .collect();
-        let verified = sig.verify_arbitrary_msg(&msg_bytes, &key).is_ok();
+        let verified = ed25519_verify_signature(&key, &sig, &msg_bytes).is_ok();
```

Comment:
Note that `diem-crypto` is not just a wrapper over dalek crate. We do more than just verifying the signature, for instance checking for malleable signatures as we see in `https://github.com/diem/diem/blob/main/crypto/crypto/src/ed25519.rs#L434`. 

Similarly `ed25519::Ed25519PublicKey::try_from` does more than just deserializing the key, as it rejects small order keys (it's not the same with `from_bytes`). 

Can you verify that the above properties still hold after this PR?

Reply:
Thanks for the reminder @kchalkias. I fixed the implementation to match to what is actually done in `diem-crypto`.

This stackless VM is not in the critical path of the diem blockchain execution (it is never used to execute any Diem transaction). It is merely a tool for the Move prover to test out whether specs are written correctly. Therefore, we do not anticipate a heavy use of crypto-related code because this is not what the Move prover is designed for.

However, I do understand that it is important to keep the implementations in-sync, both code quality-wise and for the benefit of consistency. So I duplicated the logic here and makes sure that both implementations match. In the long run, we might want to kill this version and just keep the `diem-crypto`.

Reply:
Yeah, once we properly restructure the stackless interpreter like what we did to the Move VM, native functions will likely be implemented in a Diem crate (most likely diem-framework) and be passed into the VM for execution. This way we no longer need to worry about depending on diem-crypto.